android/core/src/main/java/ai/djl/android/core/BitmapImageFactory.java

```diff
@@ -86,24 +86,29 @@ public Image fromImage(Object image) {
     @Override
     public Image fromNDArray(NDArray array) {
         Shape shape = array.getShape();
+        int height = (int) shape.get(1);
+        int width = (int) shape.get(2);
         if (shape.dimension() != 3) {
             throw new IllegalArgumentException("Shape should only have three dimension follow CHW");
         }
         if (shape.get(0) == 1) {
             throw new UnsupportedOperationException("Grayscale image is not supported");
+        } else if (shape.get(0) != 3) {
+            if (shape.get(2) == 3) {
+                height = (int) shape.get(0);
+                width = (int) shape.get(1);
+            } else {
+                throw new IllegalArgumentException("First or last dimension should be number of channel with value 1 or 3");
+            }
         }
-        if (NDImageUtils.isCHW(shape)) {
-            array = array.transpose(2, 0, 1);
-            shape = array.getShape();
-        }
-        int height = (int) shape.get(1);
-        int width = (int) shape.get(2);
-        int imageArea = width * height;
-        Bitmap bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
-        int[] raw = array.toType(DataType.UINT8, false).toUint8Array();
+        int finalWidth = width;
+        int finalHeight = height;
```

Comment:
Why do we introduce final width and height? Can we reuse?

Reply:
In the [BufferedImageFactory](https://github.com/JohnDoll2023/djl/blob/master/api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java#L105), the width is set to final there so I was trying to follow the same convention

Reply:
I would agree that we should reuse. The implication of having a variable named finalWidth is that there is also a non-final width and using it would not be right. But here, there are multiple paths to set the width, but it is not really changing. It may even make sense to change BufferedImageFactory as well. But for here, you can use width and height

Reply:
Ah I see why it is that way in BufferedImageFactory. The build will fail with error `local variables referenced from a lambda expression must be final or effectively final`. I will put `width` into a final variable and leave height just as height.